src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -393,37 +395,43 @@ private void AddAnonymousTypeName(INamedTypeSymbol symbol)
         /// Returns true if tuple type syntax can be used to refer to the tuple type without loss of information.
         /// For example, it cannot be used when extension tuple is using non-default friendly names. 
         /// </summary>
-        /// <param name="symbol"></param>
+        /// <param name="tupleSymbol"></param>
         /// <returns></returns>
-        private bool CanUseTupleTypeName(INamedTypeSymbol symbol)
+        private bool CanUseTupleTypeName(INamedTypeSymbol tupleSymbol)
         {
-            INamedTypeSymbol currentUnderlying = symbol.TupleUnderlyingType;
+            INamedTypeSymbol currentUnderlying = tupleSymbol.TupleUnderlyingType;
 
-            if (currentUnderlying.Arity == TupleTypeSymbol.RestPosition)
+            while (currentUnderlying.Arity == TupleTypeSymbol.RestPosition)
             {
-                do
+                tupleSymbol = (INamedTypeSymbol)currentUnderlying.TypeArguments[TupleTypeSymbol.RestPosition - 1];
+                Debug.Assert(tupleSymbol.IsTupleType);
+
+                if (HasNonDefaultTupleElementNames(tupleSymbol))
                 {
-                    var currentTuple = (INamedTypeSymbol)currentUnderlying.TypeArguments[TupleTypeSymbol.RestPosition - 1];
-                    Debug.Assert(currentTuple.IsTupleType);
+                    return false;
+                }
+
+                currentUnderlying = tupleSymbol.TupleUnderlyingType;
+            }
 
-                    var elementNames = currentTuple.TupleElementNames;
-                    if (!elementNames.IsDefault)
+            return true;
+        }
+
+        private bool HasNonDefaultTupleElementNames(INamedTypeSymbol tupleSymbol)
```

Comment:
I though we previously decided against computing this and instead relying on whether the array of names is default in the tuple symbol. Is there a reason to change?